### PR TITLE
Add typings for node-insights

### PR DESCRIPTION
**Pull request by wrumsby**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by jloveridge** [comment]
Not sure if this will present a problem to anyone but the `node-insights` module does not actually have an export called `default` as is implied in these types. It, as many other npm modules, simply exports the function directly. How would these typings affect someone regardless of their use of the `esModuleInterop` setting in `tsconfig.json`?

File: types/node-insights/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    timerInterval?: number;
+    maxPending?: number;
+    defaultEventType?: string;
+    enabled?: boolean;
+    gzip?: boolean;
+}
+
+export type CallbackFunc = (err: Error | null | undefined, responseBody: string) => void;
+
+export default class Insights {
```
Comment (armanio123) [inline types/node-insights/index.d.ts]: As @jloveridge has mentioend, the `default` should be removed if there's no default export.

File: types/node-insights/node-insights-tests.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import Insights from 'node-insights';
```
Comment (armanio123) [inline types/node-insights/node-insights-tests.ts]: Instead use `import x = require('module')` syntax. You can still use this style byt the `--allowSyntheticDefaultImports` or `--esModuleInterop` should be turned on instead.

**Review by armanio123 — CHANGES_REQUESTED** [review]
Check this [documentation](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require) for details about the default export.

**Comment by wrumsby** [comment]
@armanio123 changes made as requested.

**Review by armanio123 — APPROVED** [review]

**Pull request merged**